### PR TITLE
Updating `char_to_token` documentation to note behaviour when `trim_offsets` is True

### LINKED ISSUE #33237: RobertaTokenizer has incorrect character offsets
### System Info

- `transformers` version: 4.40.1
- Platform: macOS-14.6.1-arm64-arm-64bit
- Python version: 3.9.19
- Huggingface_hub version: 0.23.4
- Safetensors version: 0.4.3
- Accelerate version: 0.31.0
- Accelerate config: 	not found
- PyTorch version (GPU?): 2.3.1 (False)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: No
- Using distributed or parallel set-up in script?: No

### Who can help?

@ArthurZucker

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

I have byte offsets for a tagging task which I am trying to train a transformer with. Using the `Salesforce/codet5-base` tokenizer (which uses RobertaTokenizer) I tokenize the input and then call `char_to_token` to find out which tokens my byte offset lands in so I can build the label tensor. For some inputs `char_to_token` returns `None`, even when the character range in question appears in a token.

Running the following code:
```python
        test_string = """("    " +"""
        tok = AutoTokenizer.from_pretrained("Salesforce/codet5-base", local_files_only=True)

        enc = tok(text=test_string, truncation=False, add_special_tokens=False)

        char_mapping = []
        for i in range(len(enc["input_ids"])):
            char_mapping.append((i, enc.token_to_chars(i), tok.decode(enc["input_ids"][i])))
        print(f"\nchar_mapping = {char_mapping}")

        tok_mapping = []
        for i in range(len(test_string)):
            tok_mapping.append((i, enc.char_to_token(i)))
        print(f"\ntok_mapping = {tok_mapping}")
```
produces this output:
```python
char_mapping = [(0, CharSpan(start=0, end=2), '("'), 
                (1, CharSpan(start=5, end=5), '   '), 
                (2, CharSpan(start=6, end=7), ' "'), 
                (3, CharSpan(start=8, end=9), ' +')]

tok_mapping = [(0, 0), (1, 0), (2, None), (3, None), (4, None), (5, None), (6, 2), (7, None), (8, 3)]
```
Note that token 1 has a zero length character span despite it containing 3 space characters. The corresponding `char_to_token` call for those characters produces `None`. Also tokens 2 and 3 each contain 2 characters, but have single character spans, and again the corresponding `char_to_token` call for the start characters of those tokens returns `None`.

Part of my tagging task involves tagging string literals with a "string literal" label, but whenever the string contains whitespace I hit these `None` values and I can't find the correct token to tag.

### Expected behavior

`char_to_token` should return the correct token id when the token contains the specified character offset.

### PR DESCRIPTION
**Pull request by Craigacp**
# What does this PR do?

Updates the documentation for `BatchEncoding.char_to_token` to note that the return value can be `None` if `trim_offsets` is true and the char index points to a whitespace character.

Fixes #33237.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case. Discussed in #33237.

## Who can review?

@ArthurZucker @itazap

### CODE REVIEW

**Comment by itazap** [comment]
@Craigacp Nice, thank you!! 🚀

**Review by itazap — APPROVED** [review]

**Review by ArthurZucker — APPROVED** [review]
Thanks 🤗
